src/connector/src/parser/postgres.rs

```diff
@@ -464,3 +520,125 @@ fn pg_numeric_to_string(val: Option<PgNumeric>) -> Option<String> {
         None
     }
 }
+
+#[derive(Clone, Debug)]
+struct EnumString(String);
+
+impl<'a> FromSql<'a> for EnumString {
+    fn from_sql(
+        _ty: &Type,
+        raw: &'a [u8],
+    ) -> Result<Self, Box<dyn std::error::Error + 'static + Sync + Send>> {
+        Ok(EnumString(String::from_utf8_lossy(raw).into_owned()))
```

Comment:
I realized that this native postgres parser cannot fallback upstream data types that we don't support to `varchar`.
Besides `Enum`, for example if user want to map `INET` to varchar, we cannot make it.
Is it possible to apply this implementation, blindly store as a string, to other postgres data types? @KeXiangWang

Reply:
It depends on whether the type is serialized directly as utf8 string. We need to test different types to verify.

Reply:
https://github.com/risingwavelabs/risingwave/issues/16455
Just created an issue to track

Reply:
Another choice is that, when backfilling, instead of querying PG with `SELECT {} FROM {}` [here](https://github.com/risingwavelabs/risingwave/blob/737638052ce57393d4eed01093df3b5e69cd43e3/src/connector/src/source/cdc/external/postgres.rs#L211) and converting in RW, we can directly convert the data to varchar in pg. 

For example, if we find an enum column `E` is to be converted to varchar, we can directly run `SELECT E::varchar FROM`, so that we don't need to handle it in our codes.